test: test root user setup

mvo5 opened the pull request:
A continuation of https://github.com/osbuild/bootc-image-builder/pull/357 - the root user is used to do a `bootc status` inside the booted container to ensure the setup is correct. This is nicer then the previous `sudo` approach and it has the added benefit of testing the root user creation/setup during the build too.

achilleas-k commented on bib/go.mod:
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/cheggaaa/pb/v3 v3.1.5
 	github.com/google/uuid v1.6.0
-	github.com/osbuild/images v0.58.0
+	github.com/osbuild/images v0.58.1-0.20240430053854-ddf3ea9f0280
```
Is this an untagged commit ID?  I did a v0.59 release if you'd rather pull that in.

mvo5 replied:
Sure thing! Fixed now :)

achilleas-k reviewed (approved):
Splendid.  Minor comment but no biggie.

achilleas-k reviewed (changes requested):
Almost merged.  Missed this `fixup` commit.

mvo5 commented:
> Almost merged. Missed this `fixup` commit.

I removed the fixup commit now and fixed the bug in the PR that prevented the tests from working - it turns out we wrote invalid kickstart files because the keyfile read from disk contained a (not stripped) newline at the end and with the kickstart file looked somehting like:
```
sshkey --username foo "ssh-key
"
``` 
I opened https://github.com/osbuild/osbuild/pull/1772 with one idea
